### PR TITLE
samples: matter: Improve Matter memory profiling.

### PR DESCRIPTION
**Pull request by ArekBalysNordic**
Enabled NCS_SAMPLE_MATTER_SETTINGS_SHELL if CHIP_MEMORY_PROFILING is set.
Disabled SHELL_MINIMAL when the memory profiling is enabled.

### CODE REVIEW

File: samples/matter/common/src/Kconfig [hunk 1]
```diff
@@ -42,6 +42,7 @@ config NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE
 
 config NCS_SAMPLE_MATTER_SETTINGS_SHELL
 	bool "Settings shell for Matter purposes"
+	default y if CHIP_MEMORY_PROFILING
```
Comment (doublemis1) [inline samples/matter/common/src/Kconfig:45]: Can we enable this CHIP_MEMORY_PROFILING  CONFIG in the sample yamls?
in Light Bulb sample:
```
sample.matter.light_bulb.memory_profiling:
    build_only: true
    extra_args: CONFIG_CHIP_MEMORY_PROFILING=y
    platform_allow: nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp nrf7002dk_nrf5340_cpuapp
    integration_platforms:
      - nrf52840dk_nrf52840
      - nrf5340dk_nrf5340_cpuapp
      - nrf7002dk_nrf5340_cpuapp
```
and Bridge Simulated:
```
applications.matter_bridge.memory_profiling:
    build_only: true
    extra_args: CONFIG_BRIDGE_GENERIC_SWITCH_BRIDGED_DEVICE=n
      CONFIG_BRIDGE_ONOFF_LIGHT_SWITCH_BRIDGED_DEVICE=y
      CONFIG_CHIP_MEMORY_PROFILING=y
    integration_platforms:
      - nrf7002dk_nrf5340_cpuapp
    platform_allow: nrf7002dk_nrf5340_cpuapp
```
then we need to add them to the `quarantine_integration.yaml` in scripts folder

**Review by doublemis1 — CHANGES_REQUESTED** [review]
Please add configuration to sample yamls based on the comment

**Review by LuDuda — APPROVED** [review]

**Comment by ArekBalysNordic** [comment]
> Please add configuration to sample yamls based on the comment

@doublemis1 Done, please verify, especially the `quarantine_integration.yaml` file.

**Review by doublemis1 — APPROVED** [review]
LGTM!

**Pull request merged**